### PR TITLE
Add member ordering extended rule

### PR DESCRIPTION
**Pull request by incendial**
### What is the purpose of this pull request? (put an "X" next to an item)

[ ] Documentation update
[ ] Bug fix
[X] New rule
[ ] Changes an existing rule
[ ] Add autofixing to a rule
[ ] Add a CLI option
[ ] Add something to the core
[ ] Other, please explain:

### What changes did you make? (Give an overview)
Created an additional rule for member ordering (in order to test it separately and drop backward compatibility)

### CODE REVIEW

File: lib/src/obsoleted/rules/member_ordering_extended_rule.dart [hunk 1]
```diff
@@ -0,0 +1,709 @@
...
+            rawRepresentation: group,
+          );
+      }
+    }
+
+    return null;
+  }
+}
+
+class _Visitor extends RecursiveAstVisitor<List<_MemberInfo>> {
```
Comment (dkrutskikh) [inline lib/src/obsoleted/rules/member_ordering_extended_rule.dart:164]: @incendial extract visitor in separate file, and write separate tests

**Review by dkrutskikh — CHANGES_REQUESTED** [review]

File: CHANGELOG.md [hunk 2]
```diff
@@ -11,7 +11,8 @@
   * `ignore-files` to `exclude`
   * `maximum-nesting` to `maximum-nesting-level`
   * `number-of-arguments` to `number-of-parameters`
-* Update README
+* Update README.
+* Add static code diagnostic `member-ordering-extended`.
```
[the remark this answers is not in the document]
Comment (incendial) [reply CHANGELOG.md:15]: @dkrutskikh sure, but that's kinda strange to make one rule with snake-case naming and left others as is. I think that we should rename them separately.

File: lib/src/obsoleted/rules/member_ordering_extended_rule.dart (hunk 1 above)
Comment (incendial) [reply lib/src/obsoleted/rules/member_ordering_extended_rule.dart:164]: @dkrutskikh extracting visitor will require to make shared models public - don't think that is a good idea. What was the reason for that suggestion?

File: lib/src/obsoleted/rules/member_ordering_extended/config_parser.dart [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+  static final regExp = RegExp(
+    '(overridden_|protected_)?(private_|public_)?(static_)?(late_)?'
+    '(var_|final_|const_)?(nullable_)?(named_)?(factory_)?',
+  );
+
+  static List<_MemberGroup> parseOrder(Map<String, Object> config) {
+    final order = config.containsKey('order') && config['order'] is Iterable
+        ? List<String>.from(config['order'] as Iterable)
+        : <String>[
+            'public_fields',
```
Comment (dkrutskikh) [inline lib/src/obsoleted/rules/member_ordering_extended/config_parser.dart:14]: @incendial please extract to constant

File: test/obsoleted/rules/member_ordering_extended/member_ordering_extened_rule_test.dart [hunk 4]
```diff
@@ -0,0 +1,422 @@
...
+
+import 'dart:io';
+
+import 'package:analyzer/dart/analysis/utilities.dart';
+import 'package:dart_code_metrics/src/models/severity.dart';
+import 'package:dart_code_metrics/src/obsoleted/models/internal_resolved_unit_result.dart';
+import 'package:dart_code_metrics/src/obsoleted/rules/member_ordering_extended/member_ordering_extended_rule.dart';
+import 'package:test/test.dart';
+
+// ignore_for_file: no_adjacent_strings_in_list
```
Comment (dkrutskikh) [inline test/obsoleted/rules/member_ordering_extended/member_ordering_extened_rule_test.dart:11]: @incendial can we remove this ignore?

**Review by dkrutskikh — CHANGES_REQUESTED** [review]

File: lib/src/obsoleted/rules/member_ordering_extended/config_parser.dart (hunk 3 above)
Comment (incendial) [reply lib/src/obsoleted/rules/member_ordering_extended/config_parser.dart]: Done.

File: test/obsoleted/rules/member_ordering_extended/member_ordering_extened_rule_test.dart (hunk 4 above)
Comment (incendial) [reply test/obsoleted/rules/member_ordering_extended/member_ordering_extened_rule_test.dart]: Done.

**Review by dkrutskikh — APPROVED** [review]

**Pull request merged**